internal/controller/gitclient.go

```diff
@@ -0,0 +1,81 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/v56/github"
+	"net/http"
+)
+
+// The IssueClient interface defines an interface for issuers in Git, such as GitHub or GitLab.
+type IssueClient interface {
+	ListIssues(ctx context.Context, owner, repo string) ([]*github.Issue, error)
+	CreateIssue(ctx context.Context, owner, repo string, title, body string) (*github.Issue, error)
+	EditIssue(ctx context.Context, owner, repo string, issueNumber int, body string) (*github.Issue, error)
+	CloseIssue(ctx context.Context, owner, repo string, issueNumber int) (*github.Issue, error)
```

Comment:
Add Godoc to the functions (when defining interfaces then the doc should be above each function statement declared inside the interface)